eng/Versions.props

```diff
@@ -282,6 +271,21 @@
     <SystemCollectionsImmutableVersion>5.0.0</SystemCollectionsImmutableVersion>
     <SystemReflectionMetadataVersion>5.0.0</SystemReflectionMetadataVersion>
     <MicrosoftBclAsyncInterfacesVersion>5.0.0</MicrosoftBclAsyncInterfacesVersion>
+    <!--
+      SourceBuild will requires that all dependencies also be source buildable. We are referencing a
+      version of MSBuild that is not SourceBuild compatible, which makes our build incompatible. Since we only
+      use these dependencies as reference assemblies, we can opt them out of this behavior by having their
+      version variable be prefixed with `RefOnly`. This will allow us to reference these libraries and remain
+      Source Build compatible.
+    -->
+    <RefOnlyMicrosoftBuildVersion>$(RefOnlyMicrosoftBuildPackagesVersion)</RefOnlyMicrosoftBuildVersion>
+    <RefOnlyMicrosoftBuildFrameworkVersion>$(RefOnlyMicrosoftBuildPackagesVersion)</RefOnlyMicrosoftBuildFrameworkVersion>
+    <RefOnlyMicrosoftBuildRuntimeVersion>$(RefOnlyMicrosoftBuildPackagesVersion)</RefOnlyMicrosoftBuildRuntimeVersion>
+    <RefOnlyMicrosoftBuildTasksCoreVersion>$(RefOnlyMicrosoftBuildPackagesVersion)</RefOnlyMicrosoftBuildTasksCoreVersion>
+    <RefOnlySystemCollectionsImmutableVersion Condition="'$(RefOnlySystemCollectionsImmutableVersion)' == ''">$(SystemCollectionsImmutableVersion)</RefOnlySystemCollectionsImmutableVersion>
```

Comment:
Where do these variables come from? Source build?

Reply:
Yes, these are defined here: https://github.com/dotnet/installer/blob/main/src/SourceBuild/tarball/content/repos/roslyn.proj#L33

These variables are used to override the versions of these package references to the versions that we've already built from source.

Reply:
The `RefOnlyMicrosoftBuildPackagesVersion` is defined here in roslyn: https://github.com/dotnet/roslyn/blob/eba924a0b7b0f37d692bd3bf6eba8478308c2836/eng/Versions.props#L34

More in the same file: https://github.com/dotnet/roslyn/blob/eba924a0b7b0f37d692bd3bf6eba8478308c2836/eng/Versions.props#L274-L275

https://github.com/dotnet/roslyn/blob/eba924a0b7b0f37d692bd3bf6eba8478308c2836/eng/Versions.props#L228

https://github.com/dotnet/roslyn/blob/eba924a0b7b0f37d692bd3bf6eba8478308c2836/eng/Versions.props#L231

Having these be set conditionally lets us override the packages with reference-only versions to build against when we need to.

Reply:
I now realize you meant the `RefOnly*` variables. I accidentally named these incorrectly in this PR, They should be `SourceBuildLifted*`. I just corrected them. The variables are defined here during source build:

https://github.com/dotnet/installer/blob/2a0b8972b374311743350a5a7c8a597c501fd22e/src/SourceBuild/tarball/content/repos/roslyn.proj#L33-L36

Overriding these variables lets us build against our previously source-built versions.